command/debug.go

```diff
@@ -0,0 +1,683 @@
+			c.Ui.Error(fmt.Sprintf("Error querying node info: %s", err))
+			return 1
+		}
+		// Return error if no nodes are found
+		if len(nodes) == 0 {
+			c.Ui.Error(fmt.Sprintf("No node(s) with prefix %q found", id))
+			return 1
+		}
+
+		for _, n := range nodes {
+			c.nodeIDs = append(c.nodeIDs, n.ID)
+		}
+	}
+
+	// Resolve server prefixes
+	for _, id := range argNodes(serverIDs) {
+		c.serverIDs = append(c.serverIDs, id)
+	}
+
+	c.manifest = make([]string, 0)
+	c.stopCh = make(chan struct{})
+
+	// Setup the output path
+	format := "2006-01-02-150405Z"
+	c.timestamp = time.Now().UTC().Format(format)
+	stamped := "nomad-debug-" + c.timestamp
+
+	var tmp string
+	if output != "" {
+		tmp = filepath.Join(output, stamped)
+		_, err := os.Stat(tmp)
+		if !os.IsNotExist(err) {
+			c.Ui.Error("Output directory already exists")
+			return 2
+		}
+	} else {
+		tmp, err = ioutil.TempDir(os.TempDir(), stamped)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error creating tmp directory: %s", err.Error()))
+			return 2
+		}
+		defer os.RemoveAll(tmp)
+	}
+
+	c.collectDir = tmp
+
+	err = c.collect(client)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error collecting data: %s", err.Error()))
+		return 2
+	}
+
+	c.writeManifest()
+
+	if output != "" {
+		return 0
+	}
+
+	err = TarCZF(stamped+".tar.gz", tmp)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("error creating archive: %s", err.Error()))
+		return 2
+	}
+
+	return 0
+}
+
+// collect collects data from our endpoints and writes the archive bundle
+func (c *DebugCommand) collect(client *api.Client) error {
+	// Version contains cluster meta information
+	dir := "version"
+	err := c.mkdir(dir)
+	if err != nil {
+		return err
+	}
+
+	self, err := client.Agent().Self()
+	if err != nil {
+		return fmt.Errorf("error agent self: %s", err.Error())
+	}
+	c.writeJSON(dir, "agent-self.json", self)
+
+	// Fetch data directly from consul and vault. Ignore errors
+	var consul, vault string
+
+	m, ok := self.Config["Consul"].(map[string]interface{})
+	if ok {
+		raw := m["Addr"]
+		consul, _ = raw.(string)
+		raw = m["EnableSSL"]
+		ssl, _ := raw.(bool)
+		if ssl {
+			consul = "https://" + consul
+		} else {
+			consul = "http://" + consul
+		}
+	}
+
+	m, ok = self.Config["Vault"].(map[string]interface{})
+	if ok {
+		raw := m["Addr"]
+		vault, _ = raw.(string)
+	}
+
+	c.collectConsul(dir, consul)
+	c.collectVault(dir, vault)
+
+	c.startMonitors(client)
+	c.collectPeriodic(client)
+	c.collectPprofs(client)
+
+	return nil
+}
+
+// path returns platform specific paths in the tmp root directory
+func (c *DebugCommand) path(paths ...string) string {
+	ps := []string{c.collectDir}
+	ps = append(ps, paths...)
+	return filepath.Join(ps...)
+}
+
+// mkdir creates directories in the tmp root directory
+func (c *DebugCommand) mkdir(paths ...string) error {
+	return os.MkdirAll(c.path(paths...), 0755)
+}
+
+// startMonitors starts go routines for each node and client
+func (c *DebugCommand) startMonitors(client *api.Client) {
```

Comment:
just calling out that I'd be interested to monitor (sorry) how this might effect the nodes performance wise

Reply:
I'm curious about the same thing with pprof.